src/app/scripts/cac/control/cac-control-directions.js

```diff
@@ -332,11 +326,18 @@ CAC.Control.Directions = (function (_, $, moment, Control, Geocoder, Routing, Ty
         typeaheadFrom.setValue(destinationText);
         typeaheadTo.setValue(originText);
 
-        // set on this object and validate
-        setDirections('origin', [destination.feature.geometry.y, destination.feature.geometry.x]);
-        setDirections('destination', [origin.feature.geometry.y, origin.feature.geometry.x]);
+        // have both origin and destination
+        if (directions.origin && directions.destination) {
+            // set on this object and validate
+            setDirections('origin', [destination.feature.geometry.y, destination.feature.geometry.x]);
+            setDirections('destination', [origin.feature.geometry.y, origin.feature.geometry.x]);
+        } else if (directions.origin) {
+            setDirections('destination', [origin.feature.geometry.y, origin.feature.geometry.x]);
+        } else if (directions.destination) {
+            setDirections('origin', [destination.feature.geometry.y, destination.feature.geometry.x]);
+        }
```

Comment:
Yeah, the `setValue` is doing necessary work, but it's also clearing the origin or destination from `directions` before we're done with it.  But the conditions can just use the local `origin` and `destination` variables, which is already what the `setDirections` calls are using.